src/java.base/share/classes/jdk/internal/util/SystemProps.java

```diff
@@ -63,12 +63,13 @@ private SystemProps() {}
 
         // Platform defined encoding cannot be overridden on the command line
         put(props, "sun.jnu.encoding", raw.propDefault(Raw._sun_jnu_encoding_NDX));
+        var nativeEncoding = ((raw.propDefault(Raw._file_encoding_NDX) == null)
+                ? raw.propDefault(Raw._sun_jnu_encoding_NDX)
+                : raw.propDefault(Raw._file_encoding_NDX));
+        put(props, "native.encoding", nativeEncoding);
```

Comment:
Shouldn't native.encoding be biased toward sun.jnu.encoding rather than file.encoding? Or maybe you'll change it when preparing the changes for JEP 400?

Reply:
`native.encoding` preserves the encoding that current `Charset.defaultCharset()` is returning, which is based on `file.encoding`. So I believe the current implementation is correct. If it is biased toward `sun.jnu.encoding`, it would be problematic especially on Windows where that represents `system locale`.

Reply:
Okay for now but the value of file.encoding will change to "UTF-8" so will be different to native.encoding.

Reply:
Yes. That will be covered by JEP400.